### PR TITLE
Adds a button_type similar to default but with no border to support menu-buttons, link-buttons, icon-buttons etc.

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Fixes https://github.com/bokeh/bokeh/issues/10505

Adds a button_type `stealth` with no border to support menu-buttons, link-buttons, icon-buttons etc.

For example like

image

I chose the name "stealth" as its a "cool" name and used by the Fast Design components I'm currently wrapping. Alternatives might be `flat` or `neutral` which might be more in line with the existing `button_type` names.

As far as I can see the documentation is autogenerated, so I have not changed anything there. I just searched the code for "default" and "primary" and updated where nescessary.

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
Currently when the `stealth` button is clicked it looks like

image

I.e. there is some shadow.

There should not be. I don't know how to remove it. But there should be some kind of indication that the button is clicked.

Inspiration can be found here:

- https://getbootstrap.com/docs/4.0/components/buttons/
- MWC: https://glitch.com/~mwc-demos
- https://explore.fast.design/components/fast-button

**Comment by MarcSkovMadsen** [comment]
At getbootstrap I saw that the name for a light button without borders is `light`. Maybe that should be the name?

**Comment by MarcSkovMadsen** [comment]
I'm my little panel app I'm using

image

and

```css
.bk-btn-stealth .bk-btn-default {
    border-width: 0px;
}

.bk-btn-stealth .bk-btn-default:hover, .bk-btn-stealth .bk-btn-default:focus {
            border-width: 1px;
            color: #212529;
            background-color: #e2e6ea;
            border-color: #dae0e5;
}

.bk-btn-stealth .bk-btn-default:active, .bk-btn-stealth .bk-btn-default.bk-active  {
            border-width: 1px;
            color: #212529;
            background-color: #dae0e5;
            border-color: #d3d9df;
            box-shadow: none;
}
```

Which works

image

**hover**

image

**active**

image

**Comment by MarcSkovMadsen** [comment]
I've renamed from `stealth` `button_type` to `light` to conform with bootstrap naming convention.

File: debug.log [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[0919/064229.136:ERROR:registration_protocol_win.cc(103)] CreateFile: The system cannot find the file specified. (0x2)
+[0919/064240.868:ERROR:registration_protocol_win.cc(103)] CreateFile: The system cannot find the file specified. (0x2)
+[0919/064315.714:ERROR:registration_protocol_win.cc(103)] CreateFile: The system cannot find the file specified. (0x2)
+[0919/072904.246:ERROR:registration_protocol_win.cc(103)] CreateFile: The system cannot find the file specified. (0x2)
+[0919/072909.486:ERROR:registration_protocol_win.cc(103)] CreateFile: The system cannot find the file specified. (0x2)
```
Comment (philippjfr) [inline debug.log:5]: Remove this?

**Comment by philippjfr** [comment]
Not sure about the naming but I've definitely wanted this in the past and hacked it with custom CSS.

**Comment by MarcSkovMadsen** [comment]
I’ve wanted this so many times 😏

**Comment by bryevdv** [comment]
I think the win-3.8 failure is due to some dependency that has changed in some way

**Comment by bryevdv** [comment]
This LGTM once the errant debug log file is removed (also please merge/rebase on latest `branch-2.3` to get the win/3.8 job passing)

File: debug.log (hunk 1 above)
Comment (mattpap) [reply debug.log:5]: `git add -u` is helpful to avoid committing unnecessary files.

File: bokehjs/src/less/buttons.less [hunk 2]
```diff
@@ -92,6 +92,14 @@
     .button-variant(@btn-danger-color; @btn-danger-bg; @btn-danger-border);
   }
 
+  .bk-btn-light {
+    border-width: 0px;
```
Comment (mattpap) [inline bokehjs/src/less/buttons.less:96]: Setting border width will affect layout (both CSS and bokeh's) and make components not align (subtly, but still). One should keep the width intact and set `border-color` to `transparent`.

Comment (MarcSkovMadsen) [reply bokehjs/src/less/buttons.less]: I've pushed a fix.

File: debug.log (hunk 1 above)
Comment (MarcSkovMadsen) [reply debug.log]: I've pushed a fix which removes this file.

**Comment by MarcSkovMadsen** [comment]
I've updated to this

image

It looks like

#### No Hover

image

#### Hover

image

#### Hover and Active

image

It's not inline with the current version of Bootstrap I linked to above (it has no shadow) but I think it is inline with the current styling of bokeh widgets. And it's fine for me.

Let me know if there is more I should do. Thanks.

**Pull request merged**